doc/introduction/remote.rst

```diff
@@ -228,104 +154,122 @@ array([[0, 0, 1, 0, 1, 0, 1, 0],
        [0, 0, 0, 0, 0, 1, 0, 0],
        [0, 0, 1, 1, 0, 2, 1, 2],
        [2, 0, 1, 0, 1, 0, 0, 0]])
->>> np.mean(results.samples.T, axis=0)
+
+The samples returned correspond to 20 measurements (or shots) of the 8 mode quantum program
+above. While the majority of modes have measured zero photons, we can see that a couple have
+detected single photons, with a few even detecting 2 or 3.
+
+By taking the average of the returned array along the shots axis, we can estimate the
+mean photon number of each mode:
+
+>>> np.mean(results.samples, axis=0)
 array([0.4 , 0.1 , 0.15, 0.05, 0.3 , 0.3 , 0.45, 0.35])
 
+We can also use the Python collections module to convert the samples into
+counts:
 
-We can convert the samples into counts using the following function:
+>>> from collections import Counter
+>>> bitstrings = [tuple(i) for i in results.samples]
+>>> counts = {k:v for k, v in Counter(bitstrings).items()}
+>>> counts[(0, 0, 0, 0, 0, 0, 0)]
+2
 
-.. code-block:: python3
+.. note::
 
-     from collections import Counter
+    The :class:`~.operation` decorator allows you to create your own Strawberry Fields
+    operation. This can make it easier to ensure that the same unitary is always
+    applied to the signal and idler modes.
 
-     def count(samples):
-          bitstrings = [tuple(i) for i in samples]
-          return {k:v for k, v in Counter(bitstrings).items()}
+    .. code-block:: python3
 
->>> samples = np.array([[0, 2], [1, 0], [0, 1], [0, 0], [0, 0], [2, 0], [0, 1], [0, 1]])
->>> counts = count(samples)
->>> print(counts)
-{(0, 2): 1, (1, 0): 1, (0, 1): 3, (0, 0): 2, (2, 0): 1}
->>> counts[(0, 0)]
-2
+        from strawberryfields.utils import operation
 
-.. _compilation:
+        @operation(4)
+        def unitary(q):
+            ops.Interferometer(U) | q
+            ops.BSgate(0.543, 0.123) | (q[2], q[0])
+            ops.Rgate(0.453) | q[1]
+            ops.MZgate(0.65, -0.54) | (q[2], q[3])
 
-Program compilation
--------------------
+        prog = sf.Program(8)
 
-In addition to using the program template above, which directly matches the physical
-layout of the hardware device, you can apply any four-mode interferometer to the pairs of modes.
+        with prog.context as q:
+            ops.S2gate(1.0) | (q[0], q[4])
+            ops.S2gate(1.0) | (q[1], q[5])
+            ops.S2gate(1.0) | (q[3], q[7])
 
-Primitive gates supported by ``chip2`` include any combination of:
+            unitary() | q[:4]
```

Comment:
Still find the switching of signatures here weird

Reply:
This is a result of the `operation` decorator. I added this to the tutorial originally a couple of months ago, as several people were asking for the best shortcut to only define the unitary once - worth removing?

Reply:
No, but worth rethinking the `operation` decorator. Possible to make it something like a `property` which can be called without brackets?

Reply:
Hmm, I don't think so. It can actually accept (optional) arguments if required, so you can do something like this:

```python
@operation
custom_op(a, b, q):
    Dgate(a) | q[0]
    Rgate(b) | q[1]

with prog.context as q:
    custom_op(a, b) | q[0, 1]
```

However, the `q` argument always disappears from the signature, as the decorator moves it to the right of the `|` operator.

Reply:
Possible to put the `q` as an argument to the decorator rather than the function?

Reply:
I'll make an issue and we can look into it :+1:

Reply:
Ah, started writing out the issue, and released it's not possible that way. The function is a _definition_, `q` is an undefined dummy variable, whereas in the decorator you are calling an already defined function with a `q` that must be defined.